src/batch/src/error.rs

```diff
@@ -127,6 +128,13 @@ pub enum BatchError {
         ParquetError,
     ),
 
+    #[error(transparent)]
```

Comment:
I'm not sure if the error messages from Postgres can be clearly distinguished from ours, given that we're using `transparent` and not including any prompt here.

Reply:
As we introduce more external system queries and directly interact with them in the `batch` crate, I'm concerned that `BatchError` might become another maintainability hell just like `ConnectorError`.

Just FYI, in #15086, we made `ConnectorError` a wrapper of `anyhow::Error` because it has too many variants (may check the documentation on `def_anyhow_wrapper` for the underlying ideas). For example, shall we introduce a `ExtSystemError` with this approach (or even reuse `ConnectorError`) and make it the only variant for all external system queries in `BatchError`?

Reply:
> Just FYI, in https://github.com/risingwavelabs/risingwave/pull/15086, we made ConnectorError a wrapper of anyhow::Error because it has too many variants (may check the documentation on def_anyhow_wrapper for the underlying ideas). For example, shall we introduce a ExtSystemError with this approach (or even reuse ConnectorError) and make it the only variant for all external system queries in BatchError?

I think this a fair point. There's quite a few tvf to be supported. And we should preemptively introduce an ExtSystemError to deal with it.

Reply:
I realize there's more external system errors like iceberg, parquet from before this PR, and not just postgres error, which was introduced in this PR.
I will handle it in a separate PR since this is a larger refactor: https://github.com/risingwavelabs/risingwave/issues/18860


```

    #[error(transparent)]
    Iceberg(
        #[from]
        #[backtrace]
        iceberg::Error,
    ),

    #[error(transparent)]
    Parquet(
        #[from]
        #[backtrace]
        ParquetError,
    ),

    #[error(transparent)]
    Postgres(
        #[from]
        #[backtrace]
        tokio_postgres::Error,
    ),

```